src/sinks/util/buffer/compression.rs

```diff
@@ -0,0 +1,154 @@
+use serde::{de, ser};
+use std::fmt;
+
+#[derive(Debug, Derivative, Copy, Clone, Eq, PartialEq)]
+#[derivative(Default)]
+pub enum Compression {
+    #[derivative(Default)]
+    None,
+    Gzip(usize),
+}
+
+impl Compression {
+    pub const fn default_gzip() -> Compression {
+        Compression::Gzip(6)
+    }
```

Comment:
I note that our default when converting to a rusoto parameter is the gzip default of `6`, but for compressed buffers we had been defaulting to "fast". This makes this default awkward here.

Reply:
Also, since the constant `6` is used in a number of places, it might be helpful to move it into a constant to make it more explicit.

Reply:
I'm not sure, but can not use values with serde? We need function for this for default value on deserialization. Once compressed buffers use another value, maybe we add `gzip_fast`/`gzip_default` instead `default_gzip`?

Reply:
Regarding the constant, I just meant to add `const GZIP_DEFAULT: u32 = 6;` or similar, and use that in place of `6`. You could use `flate2::Compression::default().level()` as the constant, except that none of those functions are marked as `const` in `flate2`.

Regarding the split default functions, I would have to see how that would work. My thinking was specifically about defaulting to 6 in the serde deserializer when just `"gzip"` is specified. Were you meaning to use `gzip_fast` there?

Reply:
Heh, they just did the `const` marking, it just isn't released yet: https://github.com/rust-lang/flate2-rs/commit/0a631bbd19873c50ab87a5d120dcec28725631a9

Reply:
@bruceg I added constants and returned back fast compression for `Buffer`.